auxin_cli/src/state.rs

```diff
@@ -739,58 +614,68 @@ impl AuxinStateManager for StateManager {
 		);
 		Ok(())
 	}
-	/// Save peer record info from all peers.
+
 	fn save_all_peer_records(&mut self, context: &AuxinContext) -> crate::Result<()> {
 		info!(
 			"Start of auxin_cli's save_all_peer_records() at {}",
 			generate_timestamp()
 		);
 		let our_path = self.get_protocol_store_path(context);
 
-		let mut recipients_path = our_path.clone();
-		let mut recipients_bk_path = recipients_path.clone();
-		recipients_path.push_str("recipients-store");
-		recipients_bk_path.push_str(".bk");
+		let recipients_path = our_path.join("recipients-store");
 
-		let mut file = OpenOptions::new()
-			.truncate(true)
-			.write(true)
-			.create(true)
-			.open(recipients_path.clone())?;
+		// Working copy
+		let recipients_bk_path = recipients_path.with_extension("bk");
+
+		if recipients_bk_path.exists() {
+			error!(
+				"Working copy of recipients-store {} already exists, \
+and cannot automatically be repaired.",
+				recipients_bk_path.display()
+			);
+			return Err("recipients-store corrupt".into());
+		}
 
 		// Save recipient store:
-		let json_recipient_structure = serde_json::to_string_pretty(&context.peer_cache)?;
 		let mut bk_file = OpenOptions::new()
 			.truncate(true)
```

Comment:
Considering this is the .bk file and we're writing the working copy to it, it might make sense to remove .truncate() here. This should be a new file every time we create it, right? Existing .bk is unexpected?